src/vi/vlogtruyen/src/eu/kanade/tachiyomi/extension/vi/vlogtruyen/VlogTruyen.kt

```diff
@@ -0,0 +1,228 @@
+        .rateLimit(1)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+        .add("X-Requested-With", "XMLHttpRequest")
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/the-loai/moi-cap-nhap/?page=$page", headers)
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        title = element.select("h3.title-commic-tab").text()
+        thumbnail_url = element.select(".image-commic-tab img.lazyload").first()?.attr("data-src")
+    }
+
+    override fun latestUpdatesNextPageSelector() = ".pagination > li.active + li"
+
+    override fun latestUpdatesSelector() = "div.content-tab ul li.commic-hover"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/the-loai/dang-hot?page=$page")
+    }
+
+    override fun popularMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+
+    override fun popularMangaNextPageSelector() = latestUpdatesNextPageSelector()
+
+    override fun popularMangaSelector() = latestUpdatesSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.select("h1.title-commic-detail").text()
+        genre = document.select(".categories-list-detail-commic > li > a").joinToString { it.text().trim(',', ' ') }
+        description = document.select("div.top-detail-manga > div.top-detail-manga-content > span.desc-commic-detail").text()
+        thumbnail_url = document.select("div.image-commic-detail > a > img").attr("data-src")
+        status = parseStatus(document.select("div.top-detail-manga > div.top-detail-manga-avatar > div.manga-status > p").firstOrNull()?.text())
+        author = document.select(".h5-drawer:contains(Tác Giả) + ul li a").joinToString { it.text() }
+    }
+
+    private fun parseStatus(status: String?) = when {
+        status == null -> SManga.UNKNOWN
+        status.contains("Đang tiến hành") -> SManga.ONGOING
+        status.contains("Đã hoàn thành") -> SManga.COMPLETED
+        status.contains("Tạm ngưng") -> SManga.ON_HIATUS
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val html = response.body.string().replace("\\", "").replace("n  ", "").substringAfter("\"chaptersHtml\":\"").substringBefore("\",")
+        val document = response.asJsoup(html)
+        val hidePaidChapters = preferences.getBoolean(KEY_HIDE_PAID_CHAPTERS, false)
+        return document.select("li, .ul-list-chaper-detail-commic li").filterNot {
+            if (hidePaidChapters) {
+                it.select("li > b").text().isNotBlank().or(!hidePaidChapters)
+            } else {
+                it.select("li > a").text().isBlank().or(false)
+            }
```

Comment:
try changing selector to: `.ul-list-chaper-detail-commic li:not(:has(> b))` or similar, to filter out paid chapters directly